internal/server/data/deviceflowauthrequest.go

```diff
@@ -0,0 +1,140 @@
+package data
+
+import (
+	"gorm.io/gorm"
+
+	"github.com/infrahq/infra/internal/server/models"
+	"github.com/infrahq/infra/uid"
+)
+
+type deviceFlowAuthRequestTable models.DeviceFlowAuthRequest
+
+func (d deviceFlowAuthRequestTable) Table() string {
+	return "device_flow_auth_requests"
+}
+
+func (d deviceFlowAuthRequestTable) Columns() []string {
+	return []string{"id", "client_id", "user_code", "device_code", "approved", "access_key_id", "expires_at", "created_at",
+		"updated_at"}
+}
+
+func (d deviceFlowAuthRequestTable) Values() []any {
+	return []any{d.ID, d.ClientID, d.UserCode, d.DeviceCode, d.Approved, d.AccessKeyID, d.ExpiresAt, d.CreatedAt, d.UpdatedAt}
+}
+
+func (d *deviceFlowAuthRequestTable) ScanFields() []any {
+	return []any{d.ID, d.ClientID, d.UserCode, d.DeviceCode, d.Approved, d.AccessKeyID, d.ExpiresAt, d.CreatedAt, d.UpdatedAt}
+}
+
+// deviceFlowAuthRequestUpdateTable is used to update the DeviceFlowAuthRequest. It excludes
+// the CreatedAt field, because that field is not part of the input to
+// UpdateDeviceFlowAuthRequest.
+type deviceFlowAuthRequestUpdateTable models.DeviceFlowAuthRequest
```

Comment:
I think we don't need this as long as there is no update , or if the updates use the existing data as a base for the update.

`data.UpdateDeviceFlowAuthRequest` doesn't appear to be called yet, so not sure if we'll need that later, or if we can remove it.